flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/BigQuerySourceSplitAssigner.java

```diff
@@ -0,0 +1,247 @@
+                optionsBuilder.setRowRestriction(readOptions.getRowRestriction());
+
+                TableReadOptions options = optionsBuilder.build();
+
+                // Start specifying the read session we want created.
+                ReadSession.Builder sessionBuilder =
+                        ReadSession.newBuilder()
+                                .setTable(srcTable)
+                                .setDataFormat(DataFormat.AVRO)
+                                .setReadOptions(options);
+
+                // Optionally specify the snapshot time.  When unspecified, snapshot time is "now".
+                if (readOptions.getSnapshotTimestampInMillis() != null) {
+                    Timestamp t =
+                            Timestamp.newBuilder()
+                                    .setSeconds(readOptions.getSnapshotTimestampInMillis() / 1000)
+                                    .setNanos(
+                                            (int)
+                                                    ((readOptions.getSnapshotTimestampInMillis()
+                                                                    % 1000)
+                                                            * 1000000))
+                                    .build();
+                    TableModifiers modifiers =
+                            TableModifiers.newBuilder().setSnapshotTime(t).build();
+                    sessionBuilder.setTableModifiers(modifiers);
+                }
+
+                // Begin building the session creation request.
+                CreateReadSessionRequest.Builder builder =
+                        CreateReadSessionRequest.newBuilder()
+                                .setParent(parent)
+                                .setReadSession(sessionBuilder)
+                                .setMaxStreamCount(readOptions.getMaxStreamCount());
+
+                // request the session
+                ReadSession session = client.createReadSession(builder.build());
+                LOG.info(
+                        "BigQuery Storage Read session, name: {},"
+                                + " estimated row count {}, estimated scanned bytes {},"
+                                + " streams count {}, expired time {} (seconds after epoch).",
+                        session.getName(),
+                        session.getEstimatedRowCount(),
+                        session.getEstimatedTotalBytesScanned(),
+                        session.getStreamsCount(),
+                        session.getExpireTime().getSeconds());
+                // get all the stream names added to the initialized state
+                remainingTableStreams.addAll(
+                        session.getStreamsList().stream()
+                                .map(stream -> stream.getName())
+                                .collect(Collectors.toList()));
+                initialized = true;
+            } catch (IOException ex) {
+                throw new RuntimeException(
```

Comment:
We should have a `BigQueryConnectorException` instead of RuntimeException

Reply:
Done